activesupport/test/testing/method_call_assertions_test.rb

```diff
@@ -87,6 +94,17 @@ def test_assert_called_with_multiple_expected_arguments
       @object << 1
       @object << 2
     end
+
+    assert_called_with(@object, :<<, [ [ [ 1 ] ], [ [ 2 ] ] ]) do
+      @object << [ 1 ]
+      @object << [ 2 ]
+    end
+  end
+
+  def test_assert_called_with_expected_arguments_as_array_and_one_non_array_object
+    assert_called_with(@object, :push, [ [ 1 ], { number: 42 } ]) do
+      @object.push([ 1 ], { number: 42 })
```

Comment:
How should `@object.push([ 1 ], [ 2 ])` behave?

Reply:
:sweat_smile: Good question. Yeah looks ambiguous to me too. Maybe it is better to leave as it is then. :thinking: